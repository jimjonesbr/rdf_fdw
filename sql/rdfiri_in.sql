\pset null NULL

SELECT '<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
SELECT 'http://www.w3.org/2001/XMLSchema#string'::rdf_iri;
SELECT 'xsd:anyURI'::rdf_iri;
SELECT 'http://example/'::rdf_iri;
SELECT '<http://example/>'::rdf_iri;
SELECT 'mailto:foo@example.com'::rdf_iri;
SELECT '"mailto:foo@example.com"'::rdf_iri;
SELECT '<mailto:foo@example.com>'::rdf_iri;
SELECT '"urn:uuid:123e4567-e89b-12d3-a456-426614174000"'::rdf_iri;
SELECT '<urn:uuid:123e4567-e89b-12d3-a456-426614174000>'::rdf_iri;
SELECT 'urn:uuid:123e4567-e89b-12d3-a456-426614174000'::rdf_iri;
SELECT '"42"^^<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
SELECT '"foo"@en'::rdf_iri;
SELECT '"foo:bar"'::rdf_iri;
SELECT 'foo:bar'::rdf_iri;
SELECT '<foo:bar>'::rdf_iri;
SELECT '"foo"'::rdf_iri;
SELECT 'foo'::rdf_iri;
SELECT '<foo>'::rdf_iri;
SELECT '"a:b:c"'::rdf_iri;
SELECT 'a:b:c'::rdf_iri;
SELECT '<a:b:c>'::rdf_iri;
SELECT '"http:/not-a-scheme"'::rdf_iri; 
SELECT 'http:/not-a-scheme'::rdf_iri;
SELECT '<http:/not-a-scheme>'::rdf_iri;
SELECT '"foo"@en'::rdf_iri;
SELECT '"42"^^<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
SELECT 'http://example.com:80/foo'::rdf_iri;
SELECT 'HTTP://example.com/'::rdf_iri;
SELECT 'http://例え.テスト/こんにちは'::rdf_iri;
SELECT 'http://example.org/\u00E9'::rdf_iri;
SELECT '"foo@en'::rdf_iri;
SELECT '""^^<http://example.org/>'::rdf_iri;
SELECT '"foo"^^bar'::rdf_iri;
SELECT '_:bnode'::rdf_iri;
SELECT ''::rdf_iri;
SELECT '""'::rdf_iri;
SELECT NULL::rdf_iri;


