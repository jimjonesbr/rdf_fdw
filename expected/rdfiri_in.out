\pset null NULL
SELECT '<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
                rdf_iri                 
----------------------------------------
 <http://www.w3.org/2001/XMLSchema#int>
(1 row)

SELECT 'http://www.w3.org/2001/XMLSchema#string'::rdf_iri;
                  rdf_iri                  
-------------------------------------------
 <http://www.w3.org/2001/XMLSchema#string>
(1 row)

SELECT 'xsd:anyURI'::rdf_iri;
   rdf_iri    
--------------
 <xsd:anyURI>
(1 row)

SELECT 'http://example/'::rdf_iri;
      rdf_iri      
-------------------
 <http://example/>
(1 row)

SELECT '<http://example/>'::rdf_iri;
      rdf_iri      
-------------------
 <http://example/>
(1 row)

SELECT 'mailto:foo@example.com'::rdf_iri;
   rdf_iri    
--------------
 <mailto:foo>
(1 row)

SELECT '"mailto:foo@example.com"'::rdf_iri;
         rdf_iri          
--------------------------
 <mailto:foo@example.com>
(1 row)

SELECT '<mailto:foo@example.com>'::rdf_iri;
         rdf_iri          
--------------------------
 <mailto:foo@example.com>
(1 row)

SELECT '"urn:uuid:123e4567-e89b-12d3-a456-426614174000"'::rdf_iri;
                     rdf_iri                     
-------------------------------------------------
 <urn:uuid:123e4567-e89b-12d3-a456-426614174000>
(1 row)

SELECT '<urn:uuid:123e4567-e89b-12d3-a456-426614174000>'::rdf_iri;
                     rdf_iri                     
-------------------------------------------------
 <urn:uuid:123e4567-e89b-12d3-a456-426614174000>
(1 row)

SELECT 'urn:uuid:123e4567-e89b-12d3-a456-426614174000'::rdf_iri;
                     rdf_iri                     
-------------------------------------------------
 <urn:uuid:123e4567-e89b-12d3-a456-426614174000>
(1 row)

SELECT '"42"^^<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
 rdf_iri 
---------
 <42>
(1 row)

SELECT '"foo"@en'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT '"foo:bar"'::rdf_iri;
  rdf_iri  
-----------
 <foo:bar>
(1 row)

SELECT 'foo:bar'::rdf_iri;
  rdf_iri  
-----------
 <foo:bar>
(1 row)

SELECT '<foo:bar>'::rdf_iri;
  rdf_iri  
-----------
 <foo:bar>
(1 row)

SELECT '"foo"'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT 'foo'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT '<foo>'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT '"a:b:c"'::rdf_iri;
 rdf_iri 
---------
 <a:b:c>
(1 row)

SELECT 'a:b:c'::rdf_iri;
 rdf_iri 
---------
 <a:b:c>
(1 row)

SELECT '<a:b:c>'::rdf_iri;
 rdf_iri 
---------
 <a:b:c>
(1 row)

SELECT '"http:/not-a-scheme"'::rdf_iri; 
       rdf_iri        
----------------------
 <http:/not-a-scheme>
(1 row)

SELECT 'http:/not-a-scheme'::rdf_iri;
       rdf_iri        
----------------------
 <http:/not-a-scheme>
(1 row)

SELECT '<http:/not-a-scheme>'::rdf_iri;
       rdf_iri        
----------------------
 <http:/not-a-scheme>
(1 row)

SELECT '"foo"@en'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT '"42"^^<http://www.w3.org/2001/XMLSchema#int>'::rdf_iri;
 rdf_iri 
---------
 <42>
(1 row)

SELECT 'http://example.com:80/foo'::rdf_iri;
           rdf_iri           
-----------------------------
 <http://example.com:80/foo>
(1 row)

SELECT 'HTTP://example.com/'::rdf_iri;
        rdf_iri        
-----------------------
 <HTTP://example.com/>
(1 row)

SELECT 'http://例え.テスト/こんにちは'::rdf_iri;
             rdf_iri             
---------------------------------
 <http://例え.テスト/こんにちは>
(1 row)

SELECT 'http://example.org/\u00E9'::rdf_iri;
           rdf_iri           
-----------------------------
 <http://example.org/\u00E9>
(1 row)

SELECT '"foo@en'::rdf_iri;
  rdf_iri   
------------
 <\"foo@en>
(1 row)

SELECT '""^^<http://example.org/>'::rdf_iri;
ERROR:  invalid IRI: empty string
LINE 1: SELECT '""^^<http://example.org/>'::rdf_iri;
               ^
DETAIL:  an IRI must be a non-empty string.
SELECT '"foo"^^bar'::rdf_iri;
 rdf_iri 
---------
 <foo>
(1 row)

SELECT '_:bnode'::rdf_iri;
 rdf_iri 
---------
 _:bnode
(1 row)

SELECT ''::rdf_iri;
ERROR:  invalid IRI: empty string
LINE 1: SELECT ''::rdf_iri;
               ^
DETAIL:  an IRI must be a non-empty string.
SELECT '""'::rdf_iri;
ERROR:  invalid IRI: empty string
LINE 1: SELECT '""'::rdf_iri;
               ^
DETAIL:  an IRI must be a non-empty string.
SELECT NULL::rdf_iri;
 rdf_iri 
---------
 NULL
(1 row)

